ironfish-cli/src/commands/accounts/repair.ts

```diff
@@ -0,0 +1,203 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import {
+  Account,
+  Assert,
+  Blockchain,
+  CurrencyUtils,
+  NodeUtils,
+  Wallet,
+  WalletDB,
+} from '@ironfish/sdk'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+
+const RESCAN_MESSAGE = 'Account must be rescanned using `accounts:rescan --reset`.'
+export default class Repair extends IronfishCommand {
+  static hidden = false
+
+  static description = `Repairs wallet database stores`
+
+  static flags = {
+    ...LocalFlags,
+  }
+
+  static args = [
+    {
+      name: 'account',
+      parse: (input: string): Promise<string> => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Name of the account to repair the database for',
+    },
+  ]
+
+  async start(): Promise<void> {
+    const { args } = await this.parse(Repair)
+
+    const node = await this.sdk.node()
+    await NodeUtils.waitForOpen(node)
+
+    const account = this.loadAccount(node.wallet, args.account)
+
+    this.log(`Repairing wallet for account ${account.name}`)
+
+    this.log('Repairing balance')
+    await this.repairBalance(account, node.wallet.walletDb, node.chain)
+
+    this.log('Repairing nullifierToNote')
+    await this.repairNullifierToNoteHash(account, node.wallet.walletDb)
+
+    this.log('Repairing sequenceToNoteHash')
+    await this.repairSequenceToNoteHash(account, node.wallet.walletDb)
+  }
+
+  private loadAccount(wallet: Wallet, accountName: string | undefined): Account {
+    if (accountName) {
+      const account = wallet.getAccountByName(accountName)
+      if (account) {
+        return account
+      }
+      throw new Error(`No account with name ${accountName}`)
+    }
+
+    const defaultAccount = wallet.getDefaultAccount()
+    if (defaultAccount) {
+      return defaultAccount
+    }
+
+    throw new Error('Could not find an account to repair.')
+  }
+
+  private async repairBalance(
+    account: Account,
+    walletDb: WalletDB,
+    chain: Blockchain,
+  ): Promise<void> {
+    let unconfirmedBalance = 0n
+
+    let noteUnspentMismatches = 0
+
+    await walletDb.db.transaction(async (tx) => {
```

Comment:
This transaction will make it so any larger account can't run this command. Do we need this transaction?

Reply:
We don't need the transaction -- we should be able to safely update each note and the balance separately.

Reply:
Removed